cluster/addons/iso-client/coreaffinity-isolator.go

```diff
@@ -21,84 +18,68 @@ const (
 )
 
-func handleSIGTERM(sigterm chan os.Signal, client *aff.EventDispatcherClient, opaque *opaq.OpaqueIntegerResourceAdvertiser) {
-	<-sigterm
-	shutdownIsolator(client, opaque)
-}
-
-func shutdownIsolator(client *aff.EventDispatcherClient, opaque *opaq.OpaqueIntegerResourceAdvertiser) {
-	unregisterRequest := &lifecycle.UnregisterRequest{
-		Name:  client.Name,
-	}
-
-	glog.Infof("Unregistering custom-isolator: %s", name)
-	if _, err := client.Unregister(client.Ctx, unregisterRequest); err != nil {
-		opaque.RemoveOpaqueResource()
-		glog.Fatalf("Failed to unregister handler: %v")
-	}
-
-	glog.Infof("Removing opque integer resources %q from node %s", opaque.Name, opaque.Node)
-	if err := opaque.RemoveOpaqueResource(); err != nil {
-		glog.Fatalf("Failed to remove opaque resources: %v", err)
-	}
-
-	glog.Infof("%s has been unregistered", name)
-	os.Exit(0)
-
-}
-
 // TODO: split it to smaller functions
 func main() {
+	// enable logging to STDERR
+	flag.Set("logtostderr", "true")
 	flag.Parse()
 	glog.Info("Starting ...")
-	topology, err := discovery.DiscoverTopology()
-	if err != nil {
-		glog.Fatalf("Cannot retrive CPU topology: %q", err)
-	}
-	glog.Infof("Detected topology: %v", topology)
 
-	opaque, err := opaq.NewOpaqueIntegerResourceAdvertiser(name, fmt.Sprintf("%d", topology.GetTotalCPUs()))
+	client, err := isolator.NewEventDispatcherClient(name, eventDispatcherAddress)
 	if err != nil {
-		glog.Errorf("Cannot create opaque resource advertiser: %v", err)
-		shutdownIsolator(nil, opaque)
-	}
-	if err = opaque.AdvertiseOpaqueResource(); err != nil {
-		glog.Errorf("Failed to advertise opaque resources: %v", err)
-		shutdownIsolator(nil, opaque)
+		// cleanup in case registration fail
+		glog.Fatalf("Failed to create eventDispatcher client: %v", err)
 	}
-
-	var wg sync.WaitGroup
-	// Starting isolatorServer
-	server := aff.NewIsolator(name, isolatorLocalAddress, topology)
-	err = server.RegisterIsolator()
+	glog.Infof("EventDispatcherClient has been created for %v", client.Name)
+	// creating proper isolator
+	coreaffinityIsolator, err := coreaffinity.New(name)
 	if err != nil {
-		glog.Errorf("Cannot register isolator: %v", err)
-		shutdownIsolator(nil, opaque)
+		glog.Fatalf("Cannot create coreaffinity isolator: %q", err)
 	}
-	wg.Add(1)
-	go server.Serve(wg)
 
-	// Sending address of local isolatorServer
-	client, err := aff.NewEventDispatcherClient(name, eventDispatcherAddress, isolatorLocalAddress)
+	// bind to socket
+	socket, err := net.Listen("tcp", isolatorLocalAddress)
 	if err != nil {
-		glog.Errorf("Cannot create eventDispatcherClient: %v", err)
-		shutdownIsolator(client, opaque)
+		glog.Fatalf("Cannot create tcp socket: %v", err)
 	}
-
-	reply, err := client.Register()
-	if err != nil {
-		glog.Errorf("Failed to register isolator: %v . Reply: %v", err, reply)
-		shutdownIsolator(client, opaque)
+	// Initializing grpc server to handle isolation requests
+	server := isolator.InitializeIsolatorServer(coreaffinityIsolator)
```

Comment:
Summarizing what we talked about offline:
- Changing this to `done, err := isolator.StartServer(coreAffinityIsolator, socket)`
- Burying the following several lines of code there, including spawning the goroutine and the call to ShutDown.